plugins/ccgx/fu-ccgx-cyacd-firmware.c

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (C) 2020 Cypress Semiconductor Corporation.
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-common.h"
+#include "fu-ccgx-cyacd-firmware.h"
```

Comment:
I'm confused why you have fu-ccgx-cyacd-firmware.c **and** fu-ccgx-cyacd-file.c

Reply:
fu-ccgx-cyacd-file.c is used in both fu-ccgx-cyacd-firmware.c and fu-ccgx-dock-bb.c.
That's why I have fu-ccgx-cyacd-file.c and u-ccgx-cyacd-firmware.c.
 -  fu-ccgx-cyacd-firmware.c do not change fw and just parse cyacd to validate it.
 -  fu-ccgx-dock-bb.c parse cyacd, read cyacd row data to write firmware.
I will check your your modification and fix overlapped parsing cyacd in  fu-ccgx-cyacd-firmware.c and fu-ccgx-dock-bb.c.